src/main/java/com/feywild/feywild/EventListener.java

```diff
@@ -59,12 +57,21 @@ public void playerTick(TickEvent.PlayerTickEvent event) {
             QuestData quests = QuestData.get(player);
             player.getInventory().items.forEach(stack -> quests.checkComplete(ItemTask.INSTANCE, stack));
             //Quest Check for Biome
-            player.getLevel().getBiomeName(player.blockPosition()).ifPresent(biome -> quests.checkComplete(BiomeTask.INSTANCE, biome.location()));
+            player.getLevel().getBiome(player.blockPosition()).is(biome -> quests.checkComplete(BiomeTask.INSTANCE, biome.location()));
             //Quest Check for Structure
-            for (CompletableTaskInfo<StructureFeature<?>, StructureFeature<?>> task : quests.getAllCurrentTasks(StructureTask.INSTANCE)) {
-                if (player.getLevel().structureFeatureManager().getStructureAt(player.blockPosition(), task.getValue()).isValid()) {
-                    task.checkComplete(task.getValue());
-                }
+            if (player.getLevel().structureFeatureManager().hasAnyStructureAt(player.blockPosition())) {
+                player.getLevel().structureFeatureManager().getAllStructuresAt(player.blockPosition()).forEach((structure, set) -> quests.checkComplete(StructureTask.INSTANCE, structure));
+
+                //PORTAL
+                //   if (player.getLevel().structureFeatureManager().getAllStructuresAt(player.blockPosition()).containsKey(FeywildDimensionConfiguredFeatures.feyCircle.value())) {
+                //      FeywildDimensionHandler.teleportToFeywild(player);
+                //   }
+
+                //  player.getLevel().structureFeatureManager().getStructureAt(player.blockPosition(), FeywildDimensionConfiguredFeatures.feyCircle.value()) != null);
+
+                //   if (player.getLevel().findNearestMapFeature(ModStructureTags.ConfiguredStructureFeatures.PORTAL_STRUCTURE, player.blockPosition(), 0, true) != null) {
+                //       FeywildDimensionHandler.teleportToFeywild(player);
+                //   }
```

Comment:
I'm looking for ways to make the teleportation to the dimension more interesting, so I wanted to teleport the player to the dimension when near the fey circle structure. It didnt work as intended since the player will be teleported when it enters the chunk and not the structure location. I do think its a interesting feature wich I want to explore more. 

I'll change it to a TODO